src/components/detail-page/download-image.tsx

```diff
@@ -1,99 +1,61 @@
 import { Trans } from "@lingui/macro";
-import { Link as MUILink, Box, Typography } from "@mui/material";
+import { Link as MUILink } from "@mui/material";
+import html2canvas from "html2canvas";
 import * as React from "react";
 
-import { useLocale } from "src/lib/use-locale";
-import { useQueryState } from "src/lib/use-query-state";
-
-import { Entity } from "../../domain/data";
-
 export type Download =
   | "map"
   | "components"
   | "evolution"
   | "distribution"
   | "comparison";
 
-interface Props {
-  entity?: Entity;
-  id?: string;
-  elementId: Download;
-  fileName: string;
-  downloadType: Download;
-}
+const getImageDataFromElement = async (elementId: string): Promise<string> => {
+  const element = document.getElementById(elementId);
+  const canvas = await html2canvas(element as HTMLElement);
+  return canvas.toDataURL("image/png");
+};
 
+/**
+ * Used to download an image of a specific element or from an element
+ * that directly defines a way to get the image data.
+ */
 export const DownloadImage = ({
-  elementId,
   fileName,
-  entity,
-  id,
-  downloadType,
-}: Props) => {
-  const locale = useLocale();
-  const [
-    {
-      period,
-      category,
-      product,
-      priceComponent,
-      municipality,
-      operator,
-      canton,
-      download,
-      cantonsOrder,
-      view,
-    },
-  ] = useQueryState();
-
-  const [origin, setOrigin] = React.useState<undefined | string>(undefined);
-
-  React.useEffect(() => {
-    setOrigin(window.location.origin);
-  }, [setOrigin]);
-
-  const constructParamsFromArray = (
-    param: string[] | undefined,
-    paramName: string
-  ): string =>
-    param
-      ? `${param.reduce((acc, cur) => acc.concat(`&${paramName}=${cur}`), "")}`
-      : "";
-
-  const periods = constructParamsFromArray(period, "period");
-  const municipalities = constructParamsFromArray(municipality, "municipality");
-  const operators = constructParamsFromArray(operator, "operator");
-  const cantons = constructParamsFromArray(canton, "canton");
-
-  const queryParams = `download=${downloadType}${municipalities}${operators}${cantons}${periods}&category=${category}&product=${product}&priceComponent=${priceComponent}&cantonsOrder=${cantonsOrder}&view=${view}`;
-
-  const url =
-    entity && id && downloadType !== "map"
-      ? `${origin}/${locale}/${entity}/${id}?${queryParams}`
-      : `${origin}/${locale}?${queryParams}`;
+  elementId,
+  getImageData,
+}: {
+  downloadType?: string;
+  fileName: string;
 
-  const downLoadUrl = `${origin}/api/screenshot?url=${encodeURIComponent(
-    url
-  )}&element=${elementId}&filename=${fileName}&deviceScaleFactor=2`;
+  // Either elementId or getImageData must be provided
+  elementId?: string;
+  getImageData?: () => Promise<string | undefined>;
+}) => {
```

Comment:
Nit: Maybe it would be good to enforce this on a component level with

```ts
{ elementId: string; getImageData?: never } | { elementId?: never; getImageData: () => Promise<string | undefined> }
```

?

Reply:
Yes, I tried yesterday but could not make it to work 🤔 

<img width="573" alt="image" src="https://github.com/user-attachments/assets/47d3c38a-f8e7-49cc-8398-72bdf05beb6a">

Reply:
I had tried also with never.

<img width="558" alt="image" src="https://github.com/user-attachments/assets/178cbdad-b050-4c72-bf44-1934b334663f">

Reply:
@ptbrowne I think it should be an `or`, not `and` operator? Does it also not work then?